mantidproject/mantid
# Make the Reflectometry GUI process runs asynchronously

Previously the algorithm execution in the GenericDataProcessorPresenter was not executed asynchronously, which causes Mantid to freeze. This PR allows execution to be performed asynchronously as well as introduce a pause button for halting processing.

**To test:**

- Code review
- Ensure unit tests run are sensible.
- Perform functional testing e.g.:
  - Check that you can operate other parts of the interface while processing runs.
  - Check that window cannot be closed without pausing processing first.
  - Check that only exclusively are the `Process` and `Pause` buttons enabled/disabled, not both.
  - Ensure when pausing processing that the last row/group are processed before fully stopping.
  - Ensure that there is no change in the actual output workspaces after processing from the old interface.

There are probably many ways I haven't thought of to break the interface or do things that have unexpected side effects. Notify me of anything that feels strange to you and I will look into it.

A simple way of obtaining runs to reduce with is to login to the ICAT catalog, set current instrument to `POLREF`, enter id `1520249` and transfer them over.



Fixes #19480.

*Release notes have been updated.*

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
